app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,34 @@ def show
     @post = @user.posts.find(params[:id])
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    @like = Like.new(author_id: params[:user_id], post_id: @post.id)
+    @like.save
+    redirect_to user_post_path
+  end
+
```

Comment:
- [OPTIONAL] This should be done in the `app/controllers/likes_controller.rb` and not from the posts_controller. Please address this 👍🏻